File: app/src/main/java/com/rollling/view/adapter/BaseStandardAdapter.java
package com.rollling.view.adapter;

import android.content.Context;


/**
 * @author zhangyao
 * @date 2017/11/26:00:17
 * @E-mail android_n@163.com
 */

public abstract class BaseStandardAdapter<T, VH extends BaseViewHolder> extends BaseAdapter<T, BaseViewHolder> {

    public static final int TYPE_EMPTY= -1;

    public boolean isEmpty;

    public Context mContext;

    public int checkPosition = -1;

    public BaseStandardAdapter(Context context) {
        super(context);
        this.mContext = context;
    }

    protected final void bindHolder(BaseViewHolder holder, int position) {
        convert((VH) holder, getDataList().get(holder.getLayoutPosition()));
    }

    abstract protected void convert(VH holder, T item);

    @Override
    public final void onBindViewHolder(BaseViewHolder holder, int position) {
        holder.itemView.setClickable(true);
        bindHolder((VH) holder, position);
    }
}

File: app/src/main/java/com/rolling/view/adapter/BaseStandardAdapter.kt
package com.rolling.view.adapter

import android.content.Context

/**
 * @author zhangyao
 * @date 2017/11/26:00:17
 * @E-mail android_n@163.com
 */
abstract class BaseStandardAdapter<T, VH : BaseViewHolder<*>?>(context: Context) : BaseAdapter<T, BaseViewHolder<*>?>(context) {
    var isEmpty = false
     var mContext = context
    var checkPosition = -1
    protected fun bindHolder(holder: VH, position: Int) {
        convert(holder as VH, dataList[position])
    }

    abstract fun convert(holder: VH, item: T)
    override fun onBindViewHolder(holder: BaseViewHolder<*>, position: Int) {
        holder.itemView.isClickable = true
        bindHolder(holder as VH, position)
    }

    companion object {
        const val TYPE_EMPTY = -1
    }

}